File: Haskell/Problem1_2.hs
module ProjectEuler.Problem1_2 where

sumMultiples :: Integral a => a -> a -> a-> a
sumMultiples a b end = calc a + calc b - calc (lcm a b)
  where
    calc m = m * n * (n + 1) `div` 2
      where
        n = (end - 1) `div` m

main :: IO ()
main = print $ sumMultiples 3 5 1000

File: Clojure/problem1_2.clj
(ns project-euler.problem1_2
  (:require [clojure.math.numeric-tower :refer [lcm]]))

(defn sum-multiples [a b end]
  (letfn [(calc [m]
            (let [n (quot (dec end) m)]
              (quot (* m n (+ n 1)) 2)))]
    (- (+ (calc a) (calc b)) (calc (lcm a b)))))

(println (sum-multiples 3 5 1000))
